pilot/pkg/model/validation_test.go

```diff
@@ -318,6 +318,14 @@ func TestLabelsValidate(t *testing.T) {
 			tags:  Labels{"key": "value"},
 			valid: true,
 		},
+		{
+			name: "bad tag",
+			tags: Labels{"_key": "value"},
+		},
+		{
+			name: "bad tag",
+			tags: Labels{"key": ".value"},
+		},
```

Comment:
Add more cases, like `key_` and also some invalid characters

Reply:
Done.